app/models/grouping.rb

```diff
@@ -740,6 +740,16 @@ def final_grade(result)
   end
 
   # Helper for populate_submissions_table.
+  # Returns the total bonus/deductions for this grouping other than late penalty.
+  def total_extra_points(result)
+    if !result.nil? && result.marking_state == Result::MARKING_STATES[:complete]
```

Comment:
I get the extra points the same way as the final grade which makes the least change. The controller gets all data by model grouping.